synapse/storage/databases/main/events.py

```diff
@@ -421,61 +414,69 @@ async def _persist_events_and_state_updates(
+                        "room_ban_redactions",
+                        {"room_id": event.room_id, "user_id": event.state_key},
+                        {
+                            "redacting_event_id": event.event_id,
+                            "redact_end_ordering": None,
+                        },
+                        {
+                            "room_id": event.room_id,
+                            "user_id": event.state_key,
+                            "redacting_event_id": event.event_id,
+                            "redact_end_ordering": None,
+                        },
                     )
-                    if room_redaction_level:
-                        if sender_level < room_redaction_level:
-                            continue
-
-                    if sender_level >= redact_level:
-                        await self.db_pool.simple_upsert(
-                            "room_ban_redactions",
-                            {"room_id": event.room_id, "user_id": event.state_key},
-                            {
-                                "redacting_event_id": event.event_id,
-                                "redact_end_ordering": None,
-                            },
-                            {
-                                "room_id": event.room_id,
-                                "user_id": event.state_key,
-                                "redacting_event_id": event.event_id,
-                                "redact_end_ordering": None,
-                            },
-                        )
 
-                        # normally the cache entry for a redacted event would be invalidated
-                        # by an arriving redaction event, but since we are not creating redaction
-                        # events we invalidate manually
-                        self.store._invalidate_local_get_event_cache_room_id(
-                            event.room_id
-                        )
+                    # normally the cache entry for a redacted event would be invalidated
+                    # by an arriving redaction event, but since we are not creating redaction
+                    # events we invalidate manually
+                    self.store._invalidate_local_get_event_cache_room_id(event.room_id)
+
+                    self.store._invalidate_async_get_event_cache_room_id(event.room_id)
 
             if new_forward_extremities:
                 self.store.get_latest_event_ids_in_room.prefill(
                     (room_id,), frozenset(new_forward_extremities)
                 )
 
+    async def _can_sender_redact(self, event: EventBase) -> bool:
+        state_filter = StateFilter.from_types([(EventTypes.PowerLevels, "")])
+        state = await self.store.get_partial_filtered_current_state_ids(
+            event.room_id, state_filter
+        )
+        pl_id = state[(EventTypes.PowerLevels, "")]
+        pl_event = await self.store.get_event(pl_id)
```

Comment:
```suggestion
        pl_event = await self.store.get_event(pl_id, allow_none=True)
```

Otherwise `get_event` will raise an `EventNotFound` exception.